pkg/agent/run.go

```diff
@@ -77,11 +78,20 @@ func Run(ctx context.Context, cfg cmds.Agent) error {
 	}
 
 	cfg.DataDir = filepath.Join(cfg.DataDir, "agent")
+	os.MkdirAll(cfg.DataDir, 0700)
 
 	if cfg.ClusterSecret != "" {
 		cfg.Token = "K10node:" + cfg.ClusterSecret
 	}
 
+	lb, err := loadbalancer.Setup(ctx, cfg)
+	if err != nil {
+		return err
+	}
+	if lb != nil {
+		cfg.ServerURL = lb.LoadBalancerServerURL()
```

Comment:
does this mean even the "kubernetes" Endpoint watcher for tunneling goes through this lb?
i suspect in some cases it might be safer to fall back to the original ServerURL.

Reply:
Yep, the watcher endpoint should use the load balancer, so that value is update as early as possible. If I have `master-1` with `worker-1` and `worker-2` joined to the cluster, add a `master-2`, and shutdown `master-1`, There should be enough information for the workers to successfully connect to `master-2` on a restart.

Right now it is a simple round robin load balancing algorithm. I would like to keep the algorithm as simple as possible, but we could easily change to use the next server only on error, and append the original server url host to our list of endpoints. That should allow the balancer to fail over to the original server url as a last resort. I see that being most helpful if using an external load balancer, if all of the masters have been removed and a new one brought up the workers would still be able to re-join the cluster. We have had some internal discussions about using gossip which might also help in that regard with cluster resiliency.